tests/components/decora_wifi/test_light.py

```diff
@@ -0,0 +1,222 @@
+        data={
+            CONF_USERNAME: TEST_USERNAME,
+            CONF_PASSWORD: TEST_PASSWORD,
+        },
+    )
+    entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    return entry
+
+
+async def test_light_platform_setup(hass: HomeAssistant):
+    """Test light platform setup."""
+    # Load Registries
+    entity_registry = er.async_get(hass)
+    device_registry = dr.async_get(hass)
+
+    # Arrange Fake API State
+    FakeDecoraWiFiIotSwitch.reset_counter()
+    FakeDecoraWiFiSession.clear_accounts()
+    FakeDecoraWiFiSession.add_account(
+        FakeDecoraWiFiAccount(
+            TEST_USERNAME, TEST_PASSWORD, switch_models=["D26HD", "DW15P"]
+        )
+    )
+    # Conduct Tests
+    with patch(
+        "homeassistant.components.decora_wifi.common.DecoraWiFiSession",
+        side_effect=FakeDecoraWiFiSession,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.Residence",
+        side_effect=FakeDecoraWiFiResidence,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.ResidentialAccount",
+        side_effect=FakeDecoraWiFiResidentialAccount,
+    ):
+        # Set up the integration
+        config_entry = await setup_platform(hass)
+
+        # Assert the entity state
+        light_1_a = hass.states.get("light.fake_switch_1")
+        assert light_1_a is not None
+        assert light_1_a.state == "on"
+        assert light_1_a.attributes["brightness"] == 255
+
+        light_2 = hass.states.get("light.fake_switch_2")
+        assert light_2 is not None
+        assert light_2.state == "on"
+
+        # Assert Entity Registry RegistryEntry state
+        entity_reg_entry_1 = entity_registry.async_get("light.fake_switch_1")
+        assert entity_reg_entry_1 is not None
+        assert (
+            entity_reg_entry_1.supported_features
+            == SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+        )
+        assert entity_reg_entry_1.unique_id == "DE-AD-BE-EF-00-01"
+        entity_reg_entry_2 = entity_registry.async_get("light.fake_switch_2")
+        assert entity_reg_entry_2 is not None
+        assert entity_reg_entry_2.supported_features == 0
+        assert entity_reg_entry_2.unique_id == "DE-AD-BE-EF-00-02"
+
+        # Assert Device Registry DeviceEntry state
+        device_reg_entry_1 = device_registry.async_get(entity_reg_entry_1.device_id)
+        assert device_reg_entry_1 is not None
+        assert device_reg_entry_1.model == "D26HD"
+        assert device_reg_entry_1.manufacturer == MANUFACTURER
+        assert device_reg_entry_1.sw_version == VERSION
+        device_reg_entry_2 = device_registry.async_get(entity_reg_entry_2.device_id)
+        assert device_reg_entry_2 is not None
+        assert device_reg_entry_2.model == "DW15P"
+
+        # Assert ConfigEntry State
+        assert config_entry.state is ConfigEntryState.LOADED
+        session = hass.data[DOMAIN][config_entry.entry_id]
```

Comment:
Please don't access `hass.data[DOMAIN]` in tests

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations